### PR TITLE
Added wishlist

### PR DESCRIPTION
**Pull request by Manpreet-hub**
###  Wishlist functionality

- Add to  Wishlist
- Remove from Wishlist

### CODE REVIEW

File: src/pages/Wishlist/WishlistProducts.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { Link } from "react-router-dom";
+import { useCart, useWishList } from "../../contexts";
+
+const WishlistProducts=()=>{
+    const {wishlistState,wishlistDispatch} =useWishList();
+    const {wishlist} =wishlistState;
+    const {cartDispatch}=useCart();
+    console.log(wishlist);
```
Comment (prathmeshjagtap) [inline src/pages/Wishlist/WishlistProducts.jsx:8]: remove console.log

**Review by prathmeshjagtap — APPROVED** [review]
code is well structured, remove the console. logs, also in your preview on the header navigation, wishlist numbers are not properly visible work on that

File: src/components/Cart/CartItem.jsx [hunk 2]
```diff
@@ -25,7 +28,9 @@ const CartItem = ({ item }) => {
                                 </div>
 
                                 <button className="btn-default cart-btn" onClick={() => cartDispatch({ type: "REMOVE_FROM_CART", payload: item._id })}>Remove from cart</button>
-                                <button className="btn-default btn-dark cart-btn">Move to wishlist</button>
+                                <button className="btn-default btn-dark cart-btn" onClick={()=> {wishlistDispatch({type:"ADD_TO_WISHLIST" ,payload: item});
+                                cartDispatch({type: "REMOVE_FROM_CART", payload:item._id})
+                            }} >Move to wishlist</button>
```
Comment (abhijeetekad) [inline src/components/Cart/CartItem.jsx:33]: Use prettier to format code.

File: src/components/ProductCard.jsx [hunk 3]
```diff
@@ -12,21 +16,40 @@ const ProductCard=({product})=>{
...
-                                payload: product
-                                })}>Remove from Cart</button>
-                            
-                        ) :(
-                            <button className="btn-default btn-primary" onClick={()=>cartDispatch({
+                        cart.some(cartItem=>cartItem.id === product.id) ?
+                        (
+                            <button className="card-btn btn-primary-solid">
+                                <Link to="/cart">Go to Cart</Link>
+                            </button>)
```
Comment (abhijeetekad) [inline src/components/ProductCard.jsx:26]: +1

File: src/components/homepage/Header.jsx [hunk 4]
```diff
@@ -16,10 +17,10 @@ const Header=()=>{
             <ul className="nav-icons">
                 <li><Link to="/login" className="navbar-links" >Login </Link></li>
                 <div className="h-space-1rem"></div>
-                <li><Link to="" className="navbar-links" >
+                <li><Link to="/wishlist" className="navbar-links" >
                         <div className="badge-container">
                             <i className="fas fa-heart">
-                                <span className="badge-icon">0</span>
+                                <span className="badge-icon">{wishlist.length}</span>
```
Comment (abhijeetekad) [inline src/components/homepage/Header.jsx:23]: Add background colour/colour and border-radius here. so that rendered value will look more visible.

**Review by abhijeetekad — APPROVED** [review]
Overall code is good, great work
I like how you have destructured context value while importing.

File: src/pages/Wishlist/WishlistProducts.jsx (hunk 1 above)
Comment (Manpreet-hub) [reply src/pages/Wishlist/WishlistProducts.jsx:8]: yes, I know that, will work on header navigation while refactoring the code.

File: src/components/Cart/CartItem.jsx (hunk 2 above)
Comment (Manpreet-hub) [reply src/components/Cart/CartItem.jsx:33]: yes, will use  prettier formatter while refactoring the code.

**Pull request merged**